### ISSUE TITLE
@blueprint.response does not respect error responses

### ISSUE THREAD

**Issue by connebs** [marshmallow-code/flask-smorest#6]
Hi! I'm using your library as a convenience wrapper and I really like it!

However, one thing I've come across is that when I use the `@blueprint.response` decorator on my view function, *all* responses are marshaled with the Schema and return a 200 response. The decorator does not respect when I've finished the request with an error code.

There are many times when I abort the request (using flask's `abort`) instead of returning data. `flask-rest-api` overrides my `abort`s and `return {errors}, 409` statements, so for any view function that returns an error, I cannot use `@blueprint.response`. This means I'm using `@blueprint.response` for some of my view functions and not for others, which makes for some very inconsistent and bug-prone code.

As far as I can think of, there isn't a reason you would ever want to marshal an error response to the schema, right? So would it be possible for flask-rest-api to only marshal the response with marshmallow if the view function returns "normal" data, but respect things with error codes, like flask's `abort`? Maybe `@blueprint.response` could include a keyword arg like `respect_errors` which defaults to True or something like that? 

Thanks again, great work!

**Comment by lafrech** [comment]
Hi. Thanks for the positive feedback.

Please `abort` with flask-rest-api's `abort`. It is a webargs overload of flask `abort` that should do the right thing.

I do that a lot in my code and I do it the way you describe. The `Schema` in `response` is only used when going through the view func the "normal" way. When aborting, the overloaded `abort` stores info in the exception and the error handler in flask_rest_api/error_handler.py returns the error messages serialized as json. Maybe this deserves a few words in the docs.

The whole thing is a bit hard to grasp because it involves error messages being added to the exception object and exception handlers registered in the Flask app. This allows mimicking the behaviour of webargs, so that when aborting with a 422 (for instance when POSTing a Pet with a reference to a nonexistent category, which cannot be detected at input validation time but only when trying to write to the DB), the response is similar to a typical webargs validation error response.

Yet, there might be something else, because using flask `abort`, you wouldn't get the error messages correctly returned, but you would at least exit the function without marshshalling with the response `Schema`.

Try flask-rest-api's `abort` and if the problem persists, post a sample code here.

No need for a new parameter, what you want is current behaviour already.

**Comment by connebs** [comment]
So I tried to switch to flask_rest_api's `abort`, and no such luck. 
But I think I've figured it out, and I think it's related to #7. Whoops.

So I've been using flask_classful's `FlaskView` in order to organize my API. As flask_rest_api is not compatible with `FlaskView`, I have been using the `blueprint` decorators, but not actually in conjunction with a `blueprint`.

It's fairly hacky, and it actually took me a while to get the two to work together, because I really liked the convenience provided by your decorators, but here's what I did.

In the `__init__.py` of my flask app, I defined a flask_rest_api `Blueprint`:

```
api_bp = Blueprint("api", __name__)
```

Then, in my flask_classful `FlaskView`s, I just imported `api_bp` whenever I needed it, and used the decorators to decorate the view functions in my `FlaskView` class.

Back in the root of my flask app, in my `create_app` function, I can then register these `FlaskView`s to the `api_bp`:

```
from .api.users import UserInterface
UserInterface.register(api_bp)
```

And then finally I register the flask_rest_api `Blueprint` with my app:

```
app.register_blueprint(api_bp)
```

Apparently this hacky method to get the two to work together is actually what is breaking my `abort`s.

Any thoughts? Thanks again!
